libs/cache/CMakeLists.txt

```diff
@@ -5,150 +5,55 @@
 
 cmake_minimum_required(VERSION 3.3.2 FATAL_ERROR)
 
-project(HPX.cache CXX)
+set(module_name cache)
+string(TOUPPER ${module_name} upper_module_name)
 
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
-
-include(HPX_AddDefinitions)
-include(HPX_Message)
-include(HPX_Option)
-
-hpx_info("  cache")
-
-hpx_option(HPX_CACHE_WITH_TESTS
-  BOOL
-  "Build HPX cache module tests. (default: ${HPX_WITH_TESTS})"
-  ${HPX_WITH_TESTS} ADVANCED
-  CATEGORY "Modules")
-
-hpx_option(HPX_CACHE_WITH_DEPRECATION_WARNINGS
-  BOOL
-  "Enable warnings for deprecated facilities. (default: ${HPX_WITH_DEPRECATION_WARNINGS})"
-  ${HPX_WITH_DEPRECATION_WARNINGS} ADVANCED
-  CATEGORY "Modules")
-
-if(HPX_CACHE_WITH_DEPRECATION_WARNINGS)
-  hpx_add_config_define_namespace(
-      DEFINE HPX_CACHE_HAVE_DEPRECATION_WARNINGS
-      NAMESPACE CACHE)
-endif()
+project(HPX.${module_name} CXX)
```

Comment:
May I suggest _not_ to use the `${module_name}` cmake variable throughout the file for concrete modules to improve readability?

Reply:
I kind of agree. The question is if we can put the `project` call inside `hpx_add_module as well. Then we would barely need to use either `${module_name}` or `cache` in the `CMakeLists.txt`. @aurianer could you check if that works?

Reply:
Actually it has been removed